api/src/main/java/uk/gov/dhsc/htbhf/claimant/entitlement/CycleEntitlementCalculator.java

```diff
@@ -20,11 +20,25 @@
     private final Integer entitlementCalculationDuration;
     private final Integer numberOfCalculationPeriods;
     private final EntitlementCalculator entitlementCalculator;
+    private final Integer weeksBeforeDueDate;
+    private final Integer weeksAfterDueDate;
 
+    /**
+     * Constructor for {@link CycleEntitlementCalculator}.
+     * @param paymentCycleDurationInDays duration of payment cycle
+     * @param numberOfCalculationPeriods number of calculation periods in cycle
+     * @param weeksBeforeDueDate number of weeks before expected due date that matches a new child to that expected due date
+     * @param weeksAfterDueDate number of weeks after expected due date that matches a new child to that expected due date
+     * @param entitlementCalculator calculates entitlement for a single calculation period
+     */
     public CycleEntitlementCalculator(@Value("${payment-cycle.cycle-duration-in-days}") Integer paymentCycleDurationInDays,
                                       @Value("${payment-cycle.number-of-calculation-periods}") Integer numberOfCalculationPeriods,
+                                      @Value("${payment-cycle.child-matched-to-pregnancy-period.weeks-before-due-date}") Integer weeksBeforeDueDate,
```

Comment:
I do think it would be worthwhile encapsulating these 4 config parameters into a bean of their own for simplification, plus we don't have 4 Integers floating around which could be misused/mistaken.

Reply:
The four integers could still be misused/mistaken, just in a different place. I'm not sure moving the integers to a separate bean really adds anything. Especially as these values are only used in this class.

Reply:
I still do think its worthwhile having a `PaymentCycleConfig` bean to nicely encapsulate these fields, plus you can also have the calculation of the `entitlementCalculationDuration` field within it, which means that you only expose that value and not the other two config values - I think its definitely worthwhile!!

Reply:
Ok, moved values into configuration bean.